CONTRIBUTING.md

```diff
@@ -65,16 +29,130 @@ forth here.
+    E.g., the
+    [USDC mainnet token mint account](https://explorer.solana.com/address/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v).
+    Use apostrophes of possession,
+    [including for inanimate objects](https://english.stackexchange.com/questions/1031/is-using-the-possessive-s-correct-in-the-cars-antenna).
+    Eg 'the account's balance' is correct.
+- Don't use 'here' links. They make the course hard to scan and
+  ['here' links are bad for SEO](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/).
+- JS/TS clients send `transactions` made from `instructions`. Onchain programs
+  have `instruction handlers` that process `instructions`. Do not refer to
+  [instruction handlers](https://solana.com/docs/terminology#instruction-handler)
+  as instructions! The reason is simple: an instruction cannot process an
+  instruction. The `multiple` template in Anchor also calls the functions
+  `handler`.
+
+Finally, run Grammarly over your PRs. While your spelling might be fine,
+Grammarly will often catch dangling words and consistencies over your prose. You
+don't have to change everything Grammarly asks, but checking your document will
+save other people time. For VScode users, there is a
+[VScode extension for Grammarly](https://marketplace.visualstudio.com/items?itemName=znck.grammarly).
+
+### Code
+
+You're writing code to be read, understood, and changed by others.
+
+We want the minimal amount of code necessary to solve the problem.
+
+- Use full names. Call a `thing` a `thing`. Don't call it a `thg` - this means
+  we don't have to say 'thing, spelled tee aitch gee' every time we talk about
+  it, and means students (who often don't speak English as a first language)
+  have to work out that 'thg' means 'thing'.
+- Avoid repetitive, copy-paste code. This helps others change the code easily,
+  as they can update and fix things in a single place. If there's some
+  boilerplate, Solana-specific code you always need
+  [make a PR to the 'helpers' repository](https://github.com/solana-developers/helpers).
+- Avoid magic numbers. Nobody should see a `+ 32` in your code and wonder what
+  the `32` means. See Anchor notes below for Anchor-specific advice.
+- Avoid asking students to clone a git repo. The idea is that students should be
+  able to create projects from scratch when they have finished the course.
+  Referring to the code students have made in previous chapters is fine.
+
+#### JS/TS
+
+We're trying to focus on Solana, not teaching JS/TS development and setup. This
+means reducing the JS/TS concepts needed to understand our demo code.
+
+- `.ts` files are run with `esrun`, which supports top-level `await`, doesn't
+  require a `tsconfig.json`, etc. There is no need for `async function main()`
+  wrappers or [IIFEs](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).
+  `await` just works. If you see these wrappers, remove them.
```

Comment:
```suggestion
- On the command line (not in the browser),`.ts` files are run on the command line with `esrun`, which supports top-level `await`, doesn't require a `tsconfig.json`, etc. There is no need for `async function main()` wrappers or [IIFEs](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). `await` just works. If you see these wrappers, remove them.
```

Reply:
I spoke with @jacobcreech yesterday and wanted to make it more explicit that this is for ..ts` files on the command line. React sadly has a whole bunch of functions that aren't async and still need IIFEs to get an async context and use `await`.